rom/laghos_utils.cpp

```diff
@@ -507,11 +507,19 @@ void ReadPDweight(std::vector<double>& pd_weight, std::string outputPath)
 void SetWindowParameters(Array2D<int> const& twparam, ROM_Options & romOptions)
 {
     const int w = romOptions.window;
+
     romOptions.dimX = min(romOptions.max_dimX, twparam(w,0));
     romOptions.dimV = min(romOptions.max_dimV, twparam(w,1));
+
+	// TODO: for the energy-conserving EQP case, if twparam(w,2) exceeds the
+	// set maximum, truncating the stored basis will lead to loss of the
+	// energy identity vector, thus violating one of the energy conservation
+	// conditions. 
     romOptions.dimE = min(romOptions.max_dimE, twparam(w,2));
```

Comment:
This function is called during the online timestepping stage, to set the window basis parameters for each time window. 

In our case, the last energy basis vector formed for each time window is the energy identity vector. If the dimension of the formed energy basis exceeds the set max value, do we truncate the stored basis to the new `dimE` set here? If we do so, then we'll remove the energy identity from the basis, thus violating the energy conservation principle.